electron-1/preload.js

```diff
@@ -0,0 +1,21 @@
+const { app } = require('electron');
+const ipc = require('electron').ipcRenderer;
+
+window.bx = {
+    getName() {
+        return new Promise(function(resolve, reject) {
+            ipc.send('gibs name');
+            ipc.on('name', (stuff, name) => {
```

Comment:
It looks to me you are attaching a new callback  in any call to `getName` or `getVersion` without detaching it, no?

This could lead to [EventEmitter memory leak](https://nodejs.org/docs/latest/api/events.html#events_emitter_setmaxlisteners_n), and those Promise could be resolved multiple times.